spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/TaskDeploymentIdNotAvailableException.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.repository;
+
+/**
+ * Thrown when the task ID is not returned during task launch.
+ *
+ * @author Ilayaperumal Gopinathan
+ */
+public class TaskDeploymentIdNotAvailableException extends RuntimeException {
```

Comment:
It seems like overkill to create a new Exception type for this as it's not expected in a normal situation as far as I know (the issue was created based on `null` being returned from a dummy implementation based on the comment here: https://github.com/spring-cloud/spring-cloud-dataflow/issues/836#issuecomment-244100768

Perhaps we can just use `IllegalStateException`?